src/pages/Editor.vue

```diff
@@ -521,8 +529,19 @@ export default {
     },
     itemMarkerTimestampDragEnd(itemIndex) {
       // invoked when the drag on the marker for an item is completed
+      var timeBeforeDragEnded = this.items[itemIndex].time;
       var itemTimestamp = this.itemTimestamps[itemIndex];
-      this.items[itemIndex]["time"] = itemTimestamp;
+
+      // check if the time after drag is valid and if not, set the item time
+      // back to the one before the drag
+      // else proceed with the new time
+      if (!this.isTimestampValid(itemTimestamp, itemIndex)) {
+        this.items[itemIndex]["time"] = timeBeforeDragEnded;
+        itemTimestamp = timeBeforeDragEnded;
```

Comment:
This is needed here.
If we remove this statement and just return after showing the dialog, the marker will not move back to it's "before" position automatically. The dialog will show up yes, but the marker will stay where it was dragged to.

This is because when we drag the marker, `itemTimestamps` has already been updated with the new time position values. That is why I'm storing the before drag ended waala value in a separate variable

And we cannot also return from the function, we need the logic that is present outside the if and else waala part, to reset all the data and computed properties according to `timeBeforeDragEnded`